types/src/error.rs

```diff
@@ -177,32 +177,44 @@ impl Error {
 	}
 }
 
-/// Error type with a special `subscription_closed` field to detect that
+/// A type with a special `subscription_closed` field to detect that
 /// a subscription has been closed to distinguish valid items produced
 /// by the server on the subscription stream from an error.
 #[derive(Deserialize, Serialize, Debug, PartialEq)]
-pub struct SubscriptionClosedError {
-	subscription_closed: String,
-	id: u64,
+pub struct SubscriptionClosed {
+	subscription_closed: SubscriptionClosedReason,
```

Comment:
```suggestion
	reason: SubscriptionClosedReason,
```

…maybe?

Reply:
it will break compatibility with clients on older versions, worth it?

Reply:
yeah, let's do it then :)